packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -378,6 +394,61 @@ class RenderEditable extends RenderBox {
     return newOffset;
   }
 
+  // Handles shortcut functionality including cut, copy, paste and select all
+  // using control + (X, C, V, A).
+  void _handleShortcuts(bool ctrl, bool xKey, bool vKey, bool cKey, bool aKey) async {
+    if (cKey && !selection.isCollapsed) {
```

Comment:
this would be cleaner as a switch over pressedKeyCode

Reply:
But then you'd have to handle the boolean `selection.isCollapsed` in the case? Doesn't seem too much cleaner to me.

Reply:
That's one of the reasons it's better. Right now, if cKey is true but the selection is collapsed, you still run the code to compare against all the other values, which is wasted CPU cycles.